### PR TITLE
Problem: when running in-cluster there is no context

### PR DESCRIPTION
**Pull request by spheromak**
Solution: detect that this is a pod in a cluster, and create a context
to use.

---
This change is [Reviewable](https://reviewable.io/reviews/pantheon-systems/kube-gce-cleanup/7)

### CODE REVIEW

File: delete-orphaned-kube-network-load-balancers.sh [hunk 1]
```diff
@@ -33,6 +33,34 @@ total=0
...
+    fi
+    echo "Detected we are running in a kube pod."
+    echo "Will run some checks before continuing."
+
+    # Lets verify that our cluster == the name that someone passed in... Just in case
+    meta_kube_cluster=$(curl http://169.254.169.254/computeMetadata/v1/instance/attributes/cluster-name -q -H "Metadata-Flavor: Google")
+
+    if [[ "$GKE_CLUSTER_NAME" != "$meta_kube_cluster" ]] ; then
+        echo "You said GKE_CLUSTER_NAME is '$GKE_CLUSTER_NAME' but Google metadata says it's '$meta_kube_cluster'"
+        echo "Incase this was a mistake we wont continue."
```
Comment (winmillwill) [inline delete-orphaned-kube-network-load-balancers.sh]: `s/Incase/In case/`

File: delete-orphaned-kube-network-load-balancers.sh [hunk 2]
```diff
@@ -33,6 +33,34 @@ total=0
...
+        exit 1
+    fi
+
+    # this is a hack to work around not haivng to do custom in/out of cluster kubectl commands
```
Comment (winmillwill) [inline delete-orphaned-kube-network-load-balancers.sh]: `s/haivng/having/`

File: delete-orphaned-kube-network-load-balancers.sh [hunk 3]
```diff
@@ -33,6 +33,34 @@ total=0
...
 verb="should be deleted"
 
+
+#######################################
+# sets up a context if we are runing inside a kube cluster
+#######################################
+in_cluster_setup() {
+    # if we are a pod in kube, this env var should exist
+    if [[ -z "$KUBERNETES_SERVICE_HOST" ]] ; then
+        return
```
Comment (winmillwill) [inline delete-orphaned-kube-network-load-balancers.sh]: Is the logic that if we don't have `$KUBERNETES_SERVICE_HOST` then the validation on line 51 doesn't make any sense? If so then we might want to change the name of this function to something with `validate` in the name or just call it something like `bail_if_gke_cluster_makes_no_sense` to capture the imperative nature.

Comment (spheromak) [reply delete-orphaned-kube-network-load-balancers.sh]: yes the env var is how we know we are a pod in a cluster.  So this is really setup_if_we_are_in_cluster.

the real thing is that this actually sets the  KUBE_CONTEXT if we are running in a cluster
the validation is  a safety check as we are effectively ignoring KUBE_CONTEXT if it is provided and we are running as a pod in a cluster.   

so really its detect_and_setup_kube_pod

Comment (winmillwill) [reply delete-orphaned-kube-network-load-balancers.sh]: reconciled out of band for any who stumble on this

**Comment by winmillwill** [comment]
👍

**Pull request merged**